eng/pipelines/pr-pipeline.yml

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# This pipeline validates PRs. It builds the repo and runs inner loop tests.
+
+trigger: none
+pr:
+  - main
+
+jobs:
+  - job: Test
+    workspace:
+      clean: all
+    pool:
+      # Use generic recent LTS.
+      vmImage: ubuntu-20.04
+    steps:
+      - template: steps/checkout-unix-task.yml
+
+      - task: GoTool@0
+        inputs:
+          version: 1.17.1
```

Comment:
nit: go 1.17.2 is already out.

Approving anyway!

Reply:
Good catch--I'll merge https://github.com/microsoft/go-infra/pull/9 first, then change this PR to use the shared `eng/pipelines/steps/init-go.yml` template that specifies 1.17.2.